src/modules/auth/api.js

```diff
@@ -27,7 +27,7 @@ export class SeaCatAuthApi {
 		this.ClientSecret = "TODO";
 		this.SeaCatAuthAPI = this.App.axiosCreate('seacat_auth');
 		this.OidcAPI = this.App.axiosCreate('openidconnect');
-
+		this.state = (Math.random() + 1).toString(36).substring(7);
```

Comment:
@Aringoaway AFAIK state should consist of random string where should be also letters (upper and lower case). So this I dont consider as a state proposed by @byewokko

Reply:
I suggest generating a crypto-secure 16-byte base64 string. ChatGPT suggests using `crypto.getRandomValues()` for this :grin:

Reply:
@Aringoaway @byewokko Btw. localstorage already uses something called `redirect_code`, which is injected to localstorage by seacat auth service (afaik) and in there is a `state` represented by this random string and also a redirectUri with expirationDate ... cant we actually use it instead of re-creating something which has been already created?

this is a example of the localStorage:
```
redirect_code: '{"h97rjv0j4hvw9qyk1":{"redirectUri":"[https://auth.test.loc/auth/api/openidconnect/authorize?respo…2Fauth.test.loc%2Fsubpath%2F%3Ftenant%3Ddefault%23%2Flibrary](https://auth.test.loc/auth/api/openidconnect/authorize?response_type=code&scope=openid+tenant%3Adefault+userinfo%3A%2A&client_id=asab-webui-auth&redirect_uri=https%3A%2F%2Fauth.test.loc%2Fsubpath%2F%3Ftenant%3Ddefault%23%2Flibrary)","expirationDate":1680528772718}}',
```

Reply:
@Aringoaway So its in SeaCat Auth UI